### ISSUE TITLE
FxCop rule CA2213 wrongly issued when IDisposables are in a List

### ISSUE THREAD

**Issue by autinerd** [dotnet/roslyn-analyzers#2029]
#### Analyzer package

Example: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

Example: v2.6.3

#### Diagnostic ID

Example: [CA2213](https://docs.microsoft.com/de-de/visualstudio/code-quality/ca2213-disposable-fields-should-be-disposed)

#### Repro steps

1. Add a IDisposable, like `private System.Timers.Timer timer = new System.Timers.Timer(1000);` to the class
2. Add `protected virtual void Dispose(bool disposing) => new List<IDisposable> { timer }.ForEach((item) => item.Dispose());` to the class

#### Expected behavior

Everything is fine

#### Actual behavior

For all items in the list, the warning is issued, because it expects every item to be called directly.

**Comment by mavasani** [comment]
This does not repro for me against the latest 2.9.xxx bits from https://dotnet.myget.org/gallery/roslyn-analyzers.

```csharp
using System;
using System.Collections.Generic;

public class C : IDisposable
{
    private System.Timers.Timer timer = new System.Timers.Timer(1000);

    public void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool disposing) => new List<IDisposable> { timer }.ForEach((item) => item.Dispose());
}
```

Commenting out the body of `Dispose(bool)` does indeed lead to CA2213 violation as expected.
